runtime/src/snapshot_utils.rs

```diff
@@ -623,11 +623,9 @@ fn is_bank_snapshot_complete(bank_snapshot_dir: impl AsRef<Path>) -> bool {
 pub fn remove_tmp_snapshot_archives(snapshot_archives_dir: impl AsRef<Path>) {
     if let Ok(entries) = std::fs::read_dir(snapshot_archives_dir) {
         for entry in entries.flatten() {
-            let file_name = entry
-                .file_name()
-                .into_string()
-                .unwrap_or_else(|_| String::new());
-            if file_name.starts_with(TMP_SNAPSHOT_ARCHIVE_PREFIX) {
+            if entry.file_name().to_str().map_or(false, |file_name| {
```

Comment:
Iterator puzzles are fun, so I gave a shot at doing this with iterators. wdyt?

```rust
        for path in entries.flatten().map(|entry| entry.path()).filter(|path| {
            path.file_name()
                .filter_map(|file_name| file_name.to_str())
                .map(|file_name| file_name.starts_with(TMP_SNAPSHOT_ARCHIVE_PREFIX))
                .unwrap_or(false)
        }) {
            let result = if path.is_dir() {
                fs_err::remove_dir_all(path)
            } else {
                fs_err::remove_file(path)
            };
            if let Err(err) = result {
                warn!("Failed to remove temporary snapshot archive: {err}");
            }
```

I only looked because I don't typically like `map_or` because the order of arguments, imo, is confusing - argument order is `or` then `map` 🤔

If we don't want as significant a change as I just suggested, I'd prefer `unwrap_or` instead of `map_or` tbh.